code/modules/mob/living/living.dm

```diff
@@ -446,8 +446,11 @@
 	return
 
 /mob/living/flash_eyes(intensity = EYE_PROTECTION_FLASH, bypass_checks, type = /atom/movable/screen/fullscreen/flash, flash_timer = 40)
-	if( bypass_checks || (get_eye_protection() < intensity && !(sdisabilities & DISABILITY_BLIND)))
-		overlay_fullscreen("flash", type)
+	if(bypass_checks || (get_eye_protection() < intensity && !(sdisabilities & DISABILITY_BLIND)))
+		if(client?.prefs?.flash_overlay_pref == FLASH_OVERLAY_DARK)
```

Comment:
this doesn't check that `type` is `/atom/movable/screen/fullscreen/flash`
I know that the `type` arg isn't modified anywhere but in silicon code, but i'd like to keep this working as intended for future-proofing sake

Reply:
A `/atom/movable/screen/fullscreen/flash` is not required only a `/atom/movable/screen/fullscreen` is, and that will cause a runtime when it tries to access screen.severity in `/mob/proc/overlay_fullscreen(category, type, severity)`. So I don't see any change necessary here.

`/mob/living/flash_eyes` ultimately just puts an `/atom/movable/screen/fullscreen` on the screen after testing eye protection and animates it out in `flash_timer` time and an animate time of 2 seconds.

Reply:
Hang on sorry, just re-read Zonespace's review.

I think what they meant was that if something in the future adds a `flash_eyes()` call with a `type` of `/atom/movable/screen/fullscreen/new_fancy_screen_effect` (for example) and the player's flash preference is set to 'dark', the `flash_overlay_pref` check will take priority and will always display a `/flash/dark`, regardless of what was passed as `type`.

If this had `istype(type, /atom/movable/screen/fullscreen/flash)` (or something to the same effect) before checking the client's preferences, then it would only do it with relevant calls.
(Maybe instead of having a default `type` argument, the proc could have an empty `type_override` argument instead?)
```
if(type_override)
	overlay_fullscreen("flash", type_override)
else
	current behaviour
```